filter/set.go

```diff
@@ -0,0 +1,54 @@
+package filter
+
+import "sync"
+
+// Set is a Filter that contains 2 or more Filters
+type Set struct {
+	filters []Filter
+}
+
+// NewSet returns a Set of Filters
+func NewSet(filters ...Filter) Filter {
+	return &Set{filters: filters}
+}
+
+// Run against an input and output channel until the input channel is closed.
+// The output channel is closed on return.
+func (s *Set) Run(in <-chan int, out chan<- int) {
```

Comment:
This looks a lot like the code in the pipeline function. Why not use this code in the pipeline as well?